### PR TITLE
fix(build): correctly compile source files against typings

### PR DESCRIPTION
**Pull request by IgorMinar**
It looks like previously we were grabbing typings from outside of the tmp/ directory.
If this ever worked, it worked only by accident.

### CODE REVIEW

File: lib/broccoli/angular2-app.js [hunk 1]
```diff
@@ -31,28 +41,36 @@ Angular2App.prototype.toTree = function() {
...
 
+
+
     if (this.options && this.options.vendorNpmFiles) {
       vendorNpmFiles = vendorNpmFiles.concat(this.options.vendorNpmFiles);
     }
 
     var tsConfigCompilerOptions = JSON.parse(fs.readFileSync('src/tsconfig.json', 'utf-8')).compilerOptions;
+    // TODO(i): kill rootFilePaths in broccoli-typescript and use tsconfig.json#files instead
+    tsConfigCompilerOptions.rootFilePaths = JSON.parse(fs.readFileSync('src/tsconfig.json', 'utf-8')).files;
```
Comment (hansl) [inline lib/broccoli/angular2-app.js]: Could you load the json only once?

File: lib/broccoli/angular2-app.js [hunk 2]
```diff
@@ -18,7 +18,17 @@ function Angular2App(defaults, options, additionalPaths) {
...
-    var sourceTree = 'src';
+    var sourceDir = 'src';
+
+    var sourceTree = new Funnel('src', {
+      destDir: 'src'
+    });
+
+    var typingsTree = new Funnel('typings', {
+      include: ['browser.d.ts', 'browser/**'],
+      destDir: 'typings'
```
Comment (hansl) [inline lib/broccoli/angular2-app.js]: So this will end up in `dist/typings`, correct?

**Comment by hansl** [comment]
Couple of nits, and we still need to talk about typings.

File: lib/broccoli/angular2-app.js (hunk 1 above)
Comment (IgorMinar) [inline lib/broccoli/angular2-app.js]: ha. yeah. this should really be a separate commit. I missed this change. thanks for pointing it out.

File: lib/broccoli/angular2-app.js (hunk 2 above)
Comment (IgorMinar) [inline lib/broccoli/angular2-app.js]: no. it's available at tmp/XXX/typings during compilation but it doesn't make it into the final output.

File: lib/broccoli/angular2-app.js (hunk 1 above)
Comment (hansl) [inline lib/broccoli/angular2-app.js]: Note; `broccoli-typescript` might be going away with the upgrade to broccoli 1.0.

**Comment by hansl** [comment]
LGTM if you remove the declaration of `sourceTree`.

File: addon/ng2/blueprints/ng2/files/src/tsconfig.json [hunk 3]
```diff
@@ -9,10 +9,16 @@
     "moduleResolution": "node",
     "noEmitOnError": true,
     "noImplicitAny": false,
-    "outDir": "../dist/",
+    "//outDir": "this option is used only during manual invocation of tsc usually while debugging",
+    "outDir": "../dist-manual/",
```
Comment (hansl) [inline addon/ng2/blueprints/ng2/files/src/tsconfig.json]: Please keep this as `../dist` as this is for tooling, more than command line usage of `tsc`.

Comment (IgorMinar) [inline addon/ng2/blueprints/ng2/files/src/tsconfig.json]: fixed in a follow up commit.
